frontend/workflows/projectCatalog/src/details/card.tsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+import { Card, ClutchError, Error, Grid, styled, Typography } from "@clutch-sh/core";
+import { LinearProgress } from "@material-ui/core";
+
+export type CardTypes = "Dynamic" | "Metadata";
+
+export interface DetailsCard {
+  type: CardTypes;
+}
+
+interface CardTitleProps {
+  title: string;
+  titleIcon?: React.ReactNode;
+  endAdornment?: React.ReactNode;
+}
+
+interface CardBodyProps {
+  children?: React.ReactNode;
+  loading?: boolean;
+  error?: ClutchError;
+}
+
+interface BaseCardProps extends CardTitleProps, CardBodyProps {
+  reloadInterval?: number;
+  autoReload?: boolean;
```

Comment:
can you add some docstrings to `reloadInterval` and `autoReload`

Reply:
Yup, added comments for the entire interface. Was planning on going back and adding documentation for everything anyways.